### PR TITLE
Feature#23 add monthly attendance report

### LINKED ISSUE #23: feature#23-add-monthly-attendance-report
As an admin, I want to get a daily attendance report to know which employees on which store worked.
As an admin, I want to get a monthly report of average hours worked by employee.
As an admin, I want to get a monthly report of absence by employee.

### PR DESCRIPTION
**Pull request by GustavoHiguera**
# Description
We implemented the daily and monthly reports that the admin can do.
We created the views and the routes for each report.

 Examples:
The changes complete 3 milestones of the kata.

  ## Screenshots
kata-06-reports

### CODE REVIEW

**Review by willcab — APPROVED** [review]
LGTM 👍

**Review by Dcastanedaan — APPROVED** [review]
Good job!!

File: time_checker/app/controllers/stores_controller.rb [hunk 1]
```diff
@@ -36,6 +36,43 @@ def destroy
...
+  end
+
+  def reports_avg_time_by_month; end
+
+  def reports_avg_time_by_month_date
+    return report_empty(:average) if params[:month] == ''
+
+    date = "#{params[:month]}-01"
+    date = date.to_date
+    start_day = date
```
Comment (OswaldoPineda) [inline time_checker/app/controllers/stores_controller.rb]: did you try something like:
```
start_day = "#{params[:month]}-01".to_date
```

File: time_checker/app/controllers/stores_controller.rb [hunk 2]
```diff
@@ -36,6 +36,43 @@ def destroy
...
+  end
+
+  def reports_abs_by_month; end
+
+  def reports_abs_by_month_date
+    return report_empty(:absence) if params[:month] == ''
+
+    date = "#{params[:month]}-01"
+    date = date.to_date
+    start_day = date
```
Comment (OswaldoPineda) [inline time_checker/app/controllers/stores_controller.rb]: did you try something like:
```
start_day = "#{params[:month]}-01".to_date
```

**Review by juperameza — APPROVED** [review]
LGTM!

File: time_checker/app/controllers/stores_controller.rb [hunk 3]
```diff
@@ -36,6 +36,43 @@ def destroy
...
+  def reports_att_by_day; end
+
+  def reports_att_by_day_date
+    date = params[:check_out].to_date
+    date ? @reports = Attendance.where(check_out: date.all_day) : report_empty(:attendance)
+  end
+
+  def reports_abs_by_month; end
+
+  def reports_abs_by_month_date
```
Comment (OswaldoPineda) [inline time_checker/app/controllers/stores_controller.rb:50]: this is basically the same than the `reports_avg_time_by_month_date` pls refactor these methods
always remember the DRY(don't repeat yourself)

File: time_checker/app/views/stores/reports.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<h1 class="mb-5"><%= @store.name %>'s reports</h1>
+<div class="d-flex justify-content-evenly">
+    <%= link_to "Attendance by Day", store_reports_att_by_day_path(@store.id), class: 'btn btn-primary'  %>
+    <%= link_to "Average time employees checked in/out by month", store_reports_abs_by_month_path(@store.id), class: 'btn btn-primary'  %>
+    <%= link_to "Average hours employees worked by month", store_reports_avg_time_by_month_path(@store.id), class: 'btn btn-primary'  %>
+</div>
```
Comment (OswaldoPineda) [inline time_checker/app/views/stores/reports.html.erb]: add a final blank line here

File: time_checker/app/views/stores/reports_avg_time_by_month.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  </p>
+
+  <p>
+    <%= f.submit "Generate report", class: 'btn btn-success my-3'%>
+  </p>
+
+  <p>
+    <td><%= link_to 'Cancel and return to reports index', store_reports_path, class: 'btn btn-primary' %></td>
+  </p>
+<% end %>
```
Comment (OswaldoPineda) [inline time_checker/app/views/stores/reports_avg_time_by_month.html.erb]: add a final blank line here

**Review by OswaldoPineda — CHANGES_REQUESTED** [review]
Just some comments about refactor

**Comment by JuanequeX** [comment]
khe rayos 5,000+ changes?

File: time_checker/app/controllers/stores_controller.rb (hunk 1 above)
Comment (kalixtosh) [reply time_checker/app/controllers/stores_controller.rb]: Refactor code

File: time_checker/app/controllers/stores_controller.rb (hunk 2 above)
Comment (kalixtosh) [reply time_checker/app/controllers/stores_controller.rb]: Refactor code

File: time_checker/app/controllers/stores_controller.rb (hunk 3 above)
Comment (kalixtosh) [reply time_checker/app/controllers/stores_controller.rb:50]: Refactor code

**Comment by GustavoHiguera** [comment]
> khe rayos 5,000+ changes?

Es que sacamos los archivos de la carpeta que genera el rails a la carpeta del repositorio.

**Comment by JuanequeX** [comment]
> > khe rayos 5,000+ changes?
> 
> Es que sacamos los archivos de la carpeta que genera el rails a la carpeta del repositorio.

and what is the reason of this change?

I think, if you will planning solved the `Feature#23 add monthly attendance report`, they only should solve this specific feature and create another PR to fix your rubocop problems.

**Comment by PudinDeveloper** [comment]
### Hi!

We did this because since these are the last few days and lately pull requests take time to be approved, we made this decision.

Although we understand that it was not the right thing to do and it will not happen again.

Also, we did not do this to solve any existing problem, but to have a good practice on our project.

**Comment by JuanequeX** [comment]
> ### Hi!
> We did this because since these are the last few days and lately pull requests take time to be approved, we made this decision.
> 
> Although we understand that it was not the right thing to do and it will not happen again.
> 
> Also, we did not do this to solve any existing problem, but to have a good practice on our project.

I don't know guys, i know that in some cases is dificult to have a review, but integrate good practices in a this way it is not a good practice

**Comment by PudinDeveloper** [comment]
> I don't know guys, i know that in some cases is dificult to have a review, but integrate good practices in a this way it is not a good practice

Thank you [JuanequeX](https://github.com/JuanequeX) for the feedback, we have done a rollback and we will do it in the correct way.

**Pull request merged**